accepted/00101-health-check-tool.md

```diff
@@ -0,0 +1,216 @@
+
+As an engineer, I get the supportconfig mostly so I need this tool to consume the provided supportconfig and help me identify any potential issues before I dive into the huge amount of logs.
+  
+
+## Expected outcome
+
+* Proactive Monitoring and Maintenance: Enable system administrators to proactively monitor the health and performance of the Uyuni infrastructure. By providing comprehensive insights into the system's status, the tool aims to help administrators detect and resolve potential issues before they escalate into significant problems.
+
+* Streamlined Debugging Process for engineers: Assist engineers and developers in quickly identifying and addressing bugs or performance issues within the Uyuni codebase. 
+
+  
+  
+
+# Detailed design
+
+[design]: #detailed-design
+  
+
+## Component Overview
+
+### Health-Check-Tool
+
+* Loki/Promtail: These two components are for log aggregation, indexing and querying. Loki acts as the centralized logging system, storing and managing logs, while Promtail is deployed on the Uyuni server to collect logs and forward them to Loki.
+
+* Grafana: Utilized for visualization, with dashboards displaying metrics, logs, and alerts, facilitating an integrated view of system health and performance.
+
+* Uyuni-health-exporter: A custom exporter to gather specific metrics related to Uyuni server health, including database, service status, and resource usage.
+
+* Alerting System: Leveraging Grafana's alerting mechanism, configured to monitor critical metrics and log patterns, triggering alerts based on predefined thresholds.
+
+* LogCLI: A command-line interface tool provided by Grafana Loki, LogCLI enables direct querying of logs stored in Loki from the terminal. It supports running LogQL queries, live tailing of logs, and viewing logs in a flexible output format. This tool is particularly useful for administrators and developers who prefer to work within a CLI environment for quick log searches, debugging, and log analysis without the need for a graphical interface. 
+
+### Saline
+
+Saline is an addition for salt used in SUSE Manager/Uyuni aimed to provide better control and visibility for states deploymend in the large scale environments. It can provide metrics related to salt events and state apply process on the minions.
+
+
+## Configuration and Deployement
+
+* Monitoring Formula Integration: Loki/Promtail and uyuni_health_exporter configurations are managed via Uyuni's monitoring formula, ensuring seamless integration and ease of deployment.
+
+* Containerized Deployment: The Uyuni-Health-Check components, including Loki, Promtail, Grafana, and the Exporter, are deployed as containers within a POD, simplifying scaling and management.
```

Comment:
Will there also be a "partial" deployment that connect to an existing Grafana instance but does not deploy its own Grafana?

Reply:
I've added some more notes to clarify this